File: design-patterns/python/State-DesignPattern/src/with_state_pattern/idle_state.py
"""Idle state - no card inserted."""
from __future__ import annotations
from .atm_state import ATMState


class IdleState(ATMState):
    def insert_card(self, atm: "ATMMachine") -> None:
        from .has_card_state import HasCardState
        print("Card inserted")
        atm.set_state(HasCardState())

    def enter_pin(self, atm: "ATMMachine", pin: int) -> None:
        print("Insert card first")

    def withdraw_cash(self, atm: "ATMMachine", amount: int) -> None:
        print("Insert card first")

    def eject_card(self, atm: "ATMMachine") -> None:
        print("No card to eject")

File: design-patterns/java/State-DesignPattern/src/WithStatePattern/IdleState.java
package WithStatePattern;

public class IdleState implements ATMState {

    public void insertCard(ATMMachine atm) {
        System.out.println("Card inserted");
        atm.setState(new HasCardState());
    }

    public void enterPin(ATMMachine atm, int pin) {
        System.out.println("Insert card first");
    }

    public void withdrawCash(ATMMachine atm, int amount) {
        System.out.println("Insert card first");
    }

    public void ejectCard(ATMMachine atm) {
        System.out.println("No card to eject");
    }
}
